### ISSUE TITLE
reportSelfClsParameterName does not cover overloads

### ISSUE THREAD

**Issue by JelleZijlstra** [microsoft/pyright#4053]
Consider this code, included in a `.pyi` file:

```python
from typing import overload

class A:
    @overload
    def f(x: int, /) -> int: ...
    @overload
    def f(x, y: str, /) -> str: ...
    def g(x): ...
```

Pyright (via Pylance 2022.10.20) reports a `reportSelfClsParameterName` only for `g`, but not for either overload of `f`. This caused pyright to miss a few mistakes in a recent typeshed PR: https://github.com/python/typeshed/pull/8879/files/900e7316a225f79b291534323af4a42bffb53f1a#r996312933

**Comment by erictraut** [comment]
Currently, pyright skips this check any time a decorator is present. Perhaps that's too conservative.

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by JelleZijlstra** [comment]
Thanks! You might also want to cover `@final` functions (and `@override`, not sure if you added support for that already).

**Comment by erictraut** [comment]
This is included in pyright 1.1.276, which I just published. It will also be included in a future release of pylance.
